irispy/sji.py

```diff
@@ -2,14 +2,13 @@
 This module provides movie tools for level 2 IRIS SJI fits file
 '''
 
-from datetime import timedelta
 import warnings
 
 import numpy as np
 from astropy.io import fits
 import astropy.units as u
 from astropy.wcs import WCS
-from sunpy.time import parse_time
```

Comment:
In sunpy 1.0, ```parse_time``` returns an astropy ```Time``` object.  So I think IRISpy should continue to use ```parse_time``` so any change in sunpy is automatically picked up by irispy.  However, the cases where a numpy array of individual ```parse_time``` outputs is being creating definitely need to be changed in the way you suggest, e.g. https://github.com/sunpy/irispy/pull/121/files#diff-c3a5b1ac9cb9013bd70b2db039f006b3R508

Reply:
I'm not convinced `parse_time` is better than `Time`, for two reasons:

* It adds an additional dependency
* A change in `parse_time` may break IRISPy since it will not be guaranteed that the new object can be added to a `TimeDelta` array

Maybe @Cadair has an opinion here?

Reply:
That's true.  Those sorts of changes to sunpy would then require a change in irispy.  The counter point would be that be that presumably irispy would want to be consistent with the sunpy's time representation.  So if irispy used ```Time``` and sunpy changed, we would want to update irispy to stay consistent with that?  This is a philosophical argument about how much separation we want to put between irispy and sunpy.  So there isn't a strictly wrong answer.

Also, is irispy's only direct dependence on sunpy for ```parse_time```?  Maybe it is!  But if not, then not using ```parse_time``` does not remove a dependency.

Reply:
Looks like we use ```sunpy.util.config``` and ```check_download_file``` from sunpy as well.  Although these are to do with downloading files and so if we really wanted to remove the sunpy dependency we could probably find other solutions...  I guess one question is, are any irispy users likely to want to avoid having sunpy in their stack?

Reply:
I spoke with @Cadair and he said he will provide a third perspective.

Reply:
Ok. I didn't mean that this would be the only dependency on SunPy, so my wording was a bit unfortunate. There are several other points where IRISPy depends on SunPy, in particular for plotting (e.g. coordinates, projections, and even colour maps). If there was a way to have `parse_time` make arrays of `Time`, that would probably be the best. But otherwise, if we want consistency between time in `extra_coords` and time in the `meta` attributes, I still think it makes more sense to go with astropy time.

Reply:
Oh!  Perhaps we in fact agree then.  In the latest versions of sunpy ```parse_time``` does exactly that.  Give it an array of time stings and it returns a sinfle astropy ```Time``` object with multiple times rather than an array of individual time objects.  So as long as the user has the latest version of sunpy they will get that behaviour.  Is this what you want?  Or is your preferred behaviour an array of ```Time``` objects?

Also, reading back at your previous comments, where is a ```Time``` object added to a ```TimeDelta``` in the code?  Could you provide a link to the line?

Reply:
`Time` is added to `TimeDelta` when creating the `time` array in `extra_coords`, [for the SJI](https://github.com/tiagopereira/irispy/blob/time_fixes/irispy/sji.py#L508) and [spectrograph](https://github.com/tiagopereira/irispy/blob/time_fixes/irispy/spectrograph.py#L570). There is no sequence of time strings, only an array of seconds. So I thought it would be more logical to create an array of `TimeDelta` (which is essentially what that array is, time deltas).

Reply:
I am not 100% sure if I need to weigh in here, but in general if `Time` can parse the input there shouldn't be any reason to use `parse_time` over `Time`. When we transitioned over to `astropy.time` in SunPy we aimed to make `parse_time(...)` equivalent to `Time(...)` but accepting of more weird time formats.

Reply:
OK.  Thanks for clarifying the philosophy of ```parse_time```.  This suggests to me that we should go with @tiagopereira's approach here.